### ISSUE TITLE
请教代码中的一个细节

### ISSUE THREAD

**Issue by NLPJCL** [Hannibal046/nanoColBERT#4]
加载数据的时候，对于doc，采用了trim_padding,我理解这里是想在训练的过程中，计算query和doc的得分时，不把doc中pad的token计算进去，但是目前这个实现，是把一个batch内的doc，从原来的pad到max_len，缩减到这个doc内实际最长的doc。
那么是不是对于这个batch内稍短的doc，还是没有达到目的呢？
还是说，这里只是为了缩短seq_len，训练的时候，doc的padding token可以算进得分，只要预测的时候不算就行了？

    @staticmethod
    def collate_fn(samples,tokenizer):

        def trim_padding(input_ids,padding_id):
            ## because we padding it to make length in the preprocess script
            ## we need to trim the padded sequences in a 2-dimensional tensor to the length of the longest non-padded sequence
            non_pad_mask = input_ids != padding_id
            non_pad_lengths = non_pad_mask.sum(dim=1)
            max_length = non_pad_lengths.max().item()
            trimmed_tensor = input_ids[:,:max_length]
            return trimmed_tensor

        queries  = [x[0] for x in samples]
        pos_docs = [x[1] for x in samples]
        neg_docs = [x[2] for x in samples]

        query_input_ids = torch.from_numpy(np.stack(queries).astype(np.int32))
        query_attention_mask = (query_input_ids != tokenizer.mask_token_id).int() ## not pad token, called *query augmentation* in the paper

        doc_input_ids = torch.from_numpy(np.stack(pos_docs+neg_docs).astype(np.int32))
        doc_input_ids = trim_padding(doc_input_ids,padding_id = tokenizer.pad_token_id)
        doc_attetion_mask = (doc_input_ids != tokenizer.pad_token_id).int()

**Comment by Hannibal046** [comment]
"计算query和doc的得分时，不把doc中pad的token计算进去", 这个是在这一步完成的:
https://github.com/Hannibal046/nanoColBERT/blob/f7d069b97697a9b33fae0208036cb3bfeab7ed11/model/modeling_colbert.py#L22-L25

Trim_padding只是把多余的padding去了

**Comment by NLPJCL** [comment]
感谢，没注意到 mask_symbol_list = [tokenizer.pad_token_id] ，我以为这部分是只mask掉标点符号呢。
